overlord/snapstate/snapstate.go

```diff
@@ -1723,16 +1729,13 @@ func Remove(st *state.State, name string, revision snap.Revision) (*state.TaskSe
 	}
 
 	if tp, _ := snapst.Type(); tp == snap.TypeApp && removeAll {
-		expiration, err := AutomaticSnapshotExpiration(st)
-		if err != nil {
-			return nil, err
-		}
-		if expiration > 0 {
-			ts, err := AutomaticSnapshot(st, name)
-			if err != nil {
+		ts, err := AutomaticSnapshot(st, name)
+		if err == nil {
+			addNext(ts)
+		} else {
+			if _, ok := err.(*ErrNothingToDo); !ok {
```

Comment:
Suggestion: I think this could be a simple value that you compare to, made with `errors.New` rather than a type coupled with a type assertion.

Reply:
yes, I meant a singleton error myself as well, that was implicit in the name, sorry if it's wasn't clear.

Reminder: ErrFoo is always a single error object.   Error types should have names like FooError.

Reply:
Done